skimage/filters/thresholding.py

```diff
@@ -711,20 +714,45 @@ def threshold_li(image, *, tolerance=None, initial_guess=None,
 
     # Stop the iterations when the difference between the
     # new and old threshold values is less than the tolerance
-    while abs(t_next - t_curr) > tolerance:
-        t_curr = t_next
-        foreground = (image > t_curr)
-        mean_fore = np.mean(image[foreground])
-        mean_back = np.mean(image[~foreground])
 
-        t_next = ((mean_back - mean_fore) /
-                  (np.log(mean_back) - np.log(mean_fore)))
+    if image.dtype.kind in 'iu':
+        hist, bin_centers = histogram(image.ravel(),
```

Comment:
```suggestion
        hist, bin_centers = histogram(image.reshape(-1),
```
Please see the **Note** section in `ravel` [doc](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html) :wink:

Reply:
Thanks, I think you have mentioned this in the past, but I had forgotten about it